### PR TITLE
fix: await coin construct before moving to `Importwallet` `MethodStep`

### PR DESCRIPTION
**Pull request by goga-m**
## Checklist

-   [x] My changes look good in both light AND dark mode
-   [x] The change is not hardcoded to a single network, but has multi-asset in mind
-   [x] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: src/domains/wallet/pages/ImportWallet/ImportWallet.tsx [hunk 1]
```diff
@@ -43,6 +43,7 @@ export const ImportWallet = () => {
 
 	const [isImporting, setIsImporting] = useState(false);
 	const [isEncrypting, setIsEncrypting] = useState(false);
+	const [isSyncingCoin, setIsSyncingcoin] = useState(false);
```
Comment (dated) [inline src/domains/wallet/pages/ImportWallet/ImportWallet.tsx:46]:
```suggestion
	const [isSyncingCoin, setIsSyncingCoin] = useState(false);
```

File: src/domains/wallet/pages/ImportWallet/ImportWallet.tsx [hunk 2]
```diff
@@ -91,28 +92,50 @@ export const ImportWallet = () => {
...
-			setActiveTab(activeTab + 1);
+				setIsEncrypting(false);
+
+				break;
+			}
+			case Step.NetworkStep: {
+				// Construct coin before moving to MethodStep.
+				// Will be used in import input validations
+				const network = getValues("network");
+				setIsSyncingcoin(true);
```
Comment (dated) [inline src/domains/wallet/pages/ImportWallet/ImportWallet.tsx:126]:
```suggestion
				setIsSyncingCoin(true);
```

File: src/domains/wallet/pages/ImportWallet/ImportWallet.tsx [hunk 3]
```diff
@@ -91,28 +92,50 @@ export const ImportWallet = () => {
...
+
+				const coin = activeProfile.coins().set(network.coin(), network.id());
+				await coin.__construct();
+
+				setIsSyncingcoin(false);
```
Comment (dated) [inline src/domains/wallet/pages/ImportWallet/ImportWallet.tsx:131]:
```suggestion
				setIsSyncingCoin(false);
```

**Comment by faustbrian** [comment]
Task linked: [CU-1je4wby [Wallet Import] Exception thrown when validating address](https://app.clickup.com/t/1je4wby)

**Pull request merged**